src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/CsharpDiagnosticWorkerComposer.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Helpers;
+using OmniSharp.Models.Diagnostics;
+using OmniSharp.Options;
+using OmniSharp.Roslyn.CSharp.Services.Diagnostics;
+using OmniSharp.Roslyn.CSharp.Workers.Diagnostics;
+using OmniSharp.Services;
+
+namespace OmniSharp.Roslyn.CSharp.Workers.Diagnostics
+{
+    // Theres several implementation of worker currently based on configuration.
+    // This will handle switching between them.
+    [Export(typeof(ICsDiagnosticWorker)), Shared]
+    public class CsharpDiagnosticWorkerComposer: ICsDiagnosticWorker
+    {
+        private readonly ICsDiagnosticWorker _implementation;
+
+        [ImportingConstructor]
+        public CsharpDiagnosticWorkerComposer(
```

Comment:
Might be interesting to add tests for this class that verify that you get the same diagnostics under either implementation (unless you already have tests that do that somewhere?)

Reply:
https://github.com/OmniSharp/omnisharp-roslyn/pull/1076/commits/42ec450843496da3d9a302ddf1852ea77c786341